_episodes/01-introduction.md

```diff
@@ -3,13 +3,241 @@ title: "Introduction"
+
+### 8. Measure Performance
+
+Once we think the network is performing well we want to measure its performance. To do this we might use the validation set of data that we put aside earlier and didn't use as part of the training process. There are many different methods available for doing this and which one is best depends on the type of task we are attempting. These metrics are often published as an indication of how well our network performs. This helps to compare it with other networks or other methods for performing the same prediction or classification task.
+
+### 9. Perform a Prediction/Classification
+
+Now that we have a trained network that performs at a level we are happy with we can go and use it on real data to perform a prediction. At this point we might want to consider publishing a file with both the architecture of our network and the weights which it has learned (assuming we didn't use a pre-trained network). This will allow others to use it as as pre-trained network for their own purposes.
+
+
+> ## Deep Learning workflow exercise
+>
+> Think about a problem you'd like to use deep learning to solve.
+> 1. What do you want a deep learning system to be able to tell you?
+> 2. What data inputs and outputs will you have?
+> 3. Do you think you'll need to train the network or will a pre-trained network be suitable?
+> 4. What data do you have to train with? What preparation will your data need? Consider both the data you are going to predict/classify from and the data you'll use to train the network.
+>
+> Discuss your answers with the group or the person next to you.
+{: .challenge}
+
+## Deep Learning Libraries
+
+There are many software libraries available for deep learning including:
+
+### TensorFlow
+
+[TensorFlow](https://www.tensorflow.org/) was developed by Google and is one of the older deep learning libraries, first released to the public in 2015. It was originally developed for the C++ programming language and doesn't always follow a "pythonic" style that might make it seem a bit different to other libraries in Python. Its been ported to many other languages including Java, JavaScript, Ruby and Swift. It works on a very low level of individual "tensors" or arrays. Its capable of much more than deep learning and offers a very versatile toolkit for tensor operations. As a result it often takes a lot more lines of code to write deep learning operations in TensorFlow than other libraries. It offers (almost) seamless integration with GPU accelerators and Google's own TPU (Tensor Processing Unit) chips that are built specially for machine learning.
```

Comment:
```suggestion
[TensorFlow](https://www.tensorflow.org/) was developed by Google and is one of the older deep learning libraries, first released to the public in 2015. It was originally developed for the C++ programming language and doesn't always follow a "pythonic" style that might make it seem a bit different to other libraries in Python. It has been ported to many other languages including Java, JavaScript, Ruby and Swift. It works on a very low level of individual "tensors" or arrays. It is capable of much more than deep learning and offers a very versatile toolkit for tensor operations. As a result it often takes a lot more lines of code to write deep learning operations in TensorFlow than other libraries. It offers (almost) seamless integration with GPU accelerators and Google's own TPU (Tensor Processing Unit) chips that are built specially for machine learning.
```

Reply:
I found the detail about "tensors or arrays" confusing here. How important is it to talk this much about TensorFlow if you will not use it for the rest of your lesson? E.g. could you cut this down to

> [TensorFlow](https://www.tensorflow.org/) was developed by Google and is one of the older deep learning libraries, ported across many languages since it was first released to the public in 2015. It is very versatile and capable of much more than deep learning but as a result it often takes a lot more lines of code to write deep learning operations in TensorFlow than in other libraries. It offers (almost) seamless integration with GPU accelerators and Google's own TPU (Tensor Processing Unit) chips that are built specially for machine learning.

Reply:
resolved in commit 3b868d7